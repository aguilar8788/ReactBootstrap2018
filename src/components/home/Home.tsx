import React, {Component} from 'react'

const initialState = {}
type State = Readonly<typeof initialState>
type Props = {}

class Home extends Component<object, State> {
	constructor(props: Props) {
		super(props)
	}

	render() {
	return(
			<div className="home">
				<h1>This is Home</h1>
			</div>
		)
	}
}

export default Home