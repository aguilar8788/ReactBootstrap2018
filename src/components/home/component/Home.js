import React, {Component} from 'react'
import PropTypes from 'prop-types'

class Home extends Component {
	constructor(props, context) {
		super(props, context)
	}
	componentDidMount() {
	}

	componentWillUnmount() {
	}


	render() {
	return(
			<div className="home">
				<h1>This is Home</h1>
			</div>
		)
	}
}

Home.propTypes = {

}

Home.contextTypes = {
	router: PropTypes.object
}

export default Home