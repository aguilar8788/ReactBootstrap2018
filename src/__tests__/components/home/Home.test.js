import React from 'react'

import {shallow, mount, render} from 'enzyme'
import renderer from 'react-test-renderer'
import Home from '../../../components/home/component/Home'

describe('Home component', () => {
	it('should render without throwing errors', () => {
		expect(shallow(<Home />).exists(<div className="Home"></div>)).toBe(true)
	})

	it('renders h1 tag', () => {
		expect(shallow(<Home />).find('h1').length).toEqual(1)
	})

	it('renders correctly', () => {
		const tree = renderer.create(<Home/>).toJSON()
		expect(tree).toMatchSnapshot()
	})
})