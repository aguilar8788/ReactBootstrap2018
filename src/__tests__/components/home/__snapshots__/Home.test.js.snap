// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders correctly 1`] = `
<div
  className="home"
>
  <h1>
    This is Home
  </h1>
</div>
`;
